core/store/migrate/migrations/0093_terra_txm.sql

```diff
@@ -26,16 +29,17 @@ BEGIN
 END
 $$;
 CREATE TABLE terra_msgs (
-	id BIGSERIAL PRIMARY KEY,
--- TODO terra_chain_id text NOT NULL REFERENCES terra_chains (id),
-	contract_id text NOT NULL,
-    msg bytea NOT NULL,
-	state text NOT NULL,
-	tx_hash text, --TODO: not null for certain states
-	created_at timestamptz NOT NULL,
-	updated_at timestamptz NOT NULL
+    id BIGSERIAL PRIMARY KEY,
+    terra_chain_id text NOT NULL REFERENCES terra_chains (id),
+    contract_id text NOT NULL,
+    raw bytea NOT NULL,
+    state text NOT NULL,
+    tx_hash text, --TODO: not null for certain states
+    created_at timestamptz NOT NULL,
+    updated_at timestamptz NOT NULL
 );
 CREATE TRIGGER notify_terra_msg_insertion AFTER INSERT ON terra_msgs FOR EACH STATEMENT EXECUTE PROCEDURE notify_terra_msg_insert();
+CREATE INDEX idx_terra_msgs_terra_chain_id ON terra_nodes (terra_chain_id);
```

Comment:
I think you probably want `(terra_chain_id, state)`. That ought to give you coverage on all the queries that I've seen (we never query state without also giving a terra_chain_id).

Reply:
One additional thought - we may want a reaper on terra txes like we do for eth_Txes. Doesn't need to be done here but something to consider.